python/pyspark/sql/types.py

```diff
@@ -276,7 +276,18 @@ def toInternal(self, dt: datetime.datetime) -> int:
     def fromInternal(self, ts: int) -> datetime.datetime:
         if ts is not None:
             # using int to avoid precision loss in float
-            return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
+            return (
+                datetime.datetime
```

Comment:
We need domain experts to introduce the semantic of python datetime.

For the timestamp (ltz) type, the semantic is: it's a physical time point, and can map to different wall clock time with different local timezone. In Java, the best match is `Instant`, but I'm not sure about python `datetime` though.